### PR TITLE
Update gold endpoint

### PR DESCRIPTION
**Pull request by Piinks**
Fixes flutter/flutter#90151

The Gold team turned down the endpoint we were using. This changes it to the new API.

This also adds more logging and error handling so that if this happens again, it will be easier to diagnose.

### CODE REVIEW

**Comment by Piinks** [comment]
Blocked on https://github.com/flutter/cocoon/pull/1246
This will need to be updated once that lands

**Comment by CaseyHillers** [comment]
Cocoon is now accepting pull requests again. To make your PR compatible,
  1. Please update the base to point to main (edit in the GitHub UI)
  2. Please rebase on tip of tree. There is now null safety support and you'll have to update your PR to be null safe. There may be some rough edges around the initial migration, so feel free to add fixes as you see fit.

**Review by CaseyHillers — APPROVED** [review]
LGTM when it's rebased with tip of tree

**Pull request merged**
